spec/models/game_spec.rb

```diff
@@ -43,4 +43,40 @@
       expect(game.white_to_move).to eq(true)
     end
   end
+
+  describe "Populate Board" do
+    it "should have correct number of pieces" do
+      expect(game.pieces.count).to eq 32
+    end
+
+    it "should be empty on the squares" do
+      expect(game.pieces.where(rank: 3..6)).to eq([])
+    end
+
+    (1..8).each do |file|
+      it "should have a white pawn at rank 2, file #{file}" do
+        white_pawn = game.pieces.find{|piece| piece.rank == 2 && piece.file == file}
```

Comment:
Instead of using `find` with a block, you should use `where` or `find_by`. In the context of this test, the results may appear to be the same, but behind the scenes the first approach will bring back the entire table from the DB and then evaluate each result in memory, while `where` or `find_by` will incorporate your constraints into the resulting SQL query and bring back only the rows that you need. In a production app with a lot of data, bringing the entire pieces table into memory may be prohibitively resource intensive.